apps/dashboard/src/main/java/com/akto/action/JiraIntegrationAction.java

```diff
@@ -376,6 +388,219 @@ private BasicDBObject buildContentDetails(String txt, String link) {
         return details;
     }
 
+    String aktoDashboardHost;
+    List<TestingIssuesId> issuesIds;
+    public String bulkCreateJiraTickets (){
+        if(issuesIds == null || issuesIds.isEmpty()){
+            addActionError("Cannot create an empty jira issue.");
+            return ERROR.toUpperCase();
+        }
+
+        if((projId == null || projId.isEmpty()) || (issueType == null || issueType.isEmpty())){
+            addActionError("Project ID or Issue Type cannot be empty.");
+            return ERROR.toUpperCase();
+        }
+
+        JiraIntegration jiraInstance = JiraIntegrationDao.instance.findOne(new BasicDBObject());
+        if(jiraInstance == null) {
+            addActionError("Jira is not integrated.");
+            return ERROR.toUpperCase();
+        }
+
+        List<JiraMetaData> jiraMetaDataList = new ArrayList<>();
+        Bson projection = Projections.include(YamlTemplate.INFO);
+        List<YamlTemplate> yamlTemplateList = YamlTemplateDao.instance.findAll(Filters.empty(), projection);
+        Map<String, Info> testSubTypeToInfoMap = new HashMap<>();
+        for(YamlTemplate yamlTemplate : yamlTemplateList) {
+            Info info = yamlTemplate.getInfo();
+            testSubTypeToInfoMap.put(info.getSubCategory(), info);
+        }
+
+        List<TestingRunResult> testingRunResultList = new ArrayList<>();
+        for(TestingIssuesId testingIssuesId : issuesIds) {
+            TestingRunIssues issue = TestingRunIssuesDao.instance.findOne(Filters.and(
+                    Filters.in("_id.testSubCategory", testingIssuesId.getTestSubCategory()),
+                    Filters.in("_id.apiInfoKey." + ApiInfo.ApiInfoKey.API_COLLECTION_ID, testingIssuesId.getApiInfoKey().getApiCollectionId()),
+                    Filters.in("_id.apiInfoKey." + ApiInfo.ApiInfoKey.URL, testingIssuesId.getApiInfoKey().getUrl()),
+                    Filters.in("_id.apiInfoKey." + ApiInfo.ApiInfoKey.METHOD, testingIssuesId.getApiInfoKey().getMethod()),
+                    Filters.exists("jiraIssueUrl", true)
+            ));
+
+            if(issue != null && (issue.getJiraIssueUrl() != null || !issue.getJiraIssueUrl().isEmpty())) {
+                addActionError("One of the selected issues has a Jira issue created for it.");
```

Comment:
doesnt matter, why stop other tickets to be formed, add filter while getting issue, jiraissueurl not exists, return count of how many created